### ISSUE TITLE
Use number operators in where filters for amount of references

### ISSUE THREAD

**Issue by etiennedi** [semi-technologies/weaviate#1101]
## Todos
* [x] built intergration test setup
* [x] support all possible operator configurations
* [x] catch invalid data type errors
* [x] add journey test

## Example

E.g.

```graphql
{
  Get {
    Things {
      Author (where:{
        operator: GreaterThan,
        path:["WroteArticles"]
        valueInt: 0
      }){
        name
        WroteArticles {
          ... on Article {
            InPublication {
              ... on Publication {
                name
              }
            }
          }
        }
      }
    }
  }
}
```
would only show Authors with more than 0 outgoing refs in the `WroteArticles` field. Or in plain english: Authors who wrote at least one article.

**Comment by bobvanluijt** [comment]
Note, resolving this is the answer for: https://stackoverflow.com/questions/60581197/is-there-a-way-to-filter-out-null-values-for-literals-and-references
